ambry-api/src/main/java/com.github.ambry/config/RouterConfig.java

```diff
@@ -147,6 +147,13 @@
   public final double routerLatencyToleranceQuantile;
 
   /**
+   * The version to use for new BlobIds.
+   */
+  @Config("router.blobid.version")
+  @Default("2")
+  public final short routerBlobidCurrentVersion;
```

Comment:
It's not a config test I'm suggesting - there doesn't seem to be a test in `PutOperation` where you change the config and it writes in the version you want it to i.e. something that would fail if the the way of generating `BlobId` in `PutOperation` if the code had never been changed.

Reply:
For an incremental change such as this it might appear that we want such a test, but we don't generally have tests for whether the config values are respected by the components. Tests do change certain configs - but generally they do that to test the functionality of the flow enabled by that change, not to test whether such a flow gets enabled at all.

For example, if we wanted to test whether the router would have any difficulty putting blobs in V3 (which I don't think we need to - Blob id construction tests suffice), then we would set the config accordingly - but not to test whether by setting the config the blob id that the router creates is actually a V3 blob id.